pallets/parachain-staking/src/lib.rs

```diff
@@ -2048,6 +2054,14 @@ pub mod pallet {
 			let actual_weight = T::WeightInfo::delegator_bond_more(
 				<DelegationScheduledRequests<T>>::get(&candidate).len() as u32,
```

Comment:
```suggestion
				if already_called_for_delegator { 0 } else { <DelegationScheduledRequests<T>>::get(&candidate).len() as u32 }
```

Reply:
I recommend this as well.

Reply:
Alternate would be to ignore the weight generated from this function and use `T::WeightInfo::delegator_bond_more(0)` in the `mint_and_compound`